libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt

```diff
@@ -111,6 +111,7 @@ class RouteRefreshController internal constructor(
 
     internal fun requestPlannedRouteRefresh(routes: List<NavigationRoute>) {
         routeRefresherResultProcessor.reset()
+        immediateRouteRefreshController.cancel()
```

Comment:
I don't think so. Otherwise we might get updated routes that are now changed.
When immediate refresh finishes, the routes might have already been changed. We'd get a race here. We'll notify RoutesObserver with old refreshed routes.